test/Rings/mpoly-test.jl

```diff
@@ -69,5 +69,26 @@ end
   I = ideal([2*x+3*y+4*z-5,3*x+4*y+5*z-2])
   @test groebner_basis(I,:degrevlex) == [y+2*z-11, 3*x+4*y+5*z-2]
   @test groebner_basis(I,:degrevlex, complete_reduction = true) == [y+2*z-11, x-z+14]
-  
+
+end
+
+@testset "Primary decomposition" begin
+  R, (x, y, z) = PolynomialRing(QQ, ["x", "y", "z"])
+  I = ideal(R, [x, y*z^2])
+  J = ideal(R, [x, y^2])
+  L = primary_decomposition(I)
+  Q = ideal(R, [R(1)])
+  @test isprime(I) == false
+  @test isprimary(I) == false
+  @test isprime(J) == false
+  @test isprimary(J) == true
+  for (q, p) in L
+    Q = intersect(Q, q)
+    @test isprimary(q)
+    @test isprime(p)
+    rq = radical(q)
+    @test Singular.equal(rq.gens.S, p.gens.S)
```

Comment:
It turns out the ideal equality is implemented properly in oscar, so you should be able to do
`@test p == radical(q)`
for these two lines. The test code should try to match what a user is going to do. We should not expect/require users to know that ideals have a member called `.gens` and that this in turn has a member called `.S`.